ui.apps/src/main/content/jcr_root/apps/core/fd/components/clientlib-core-forms-components-runtime/js/main.js

```diff
@@ -1 +1 @@
um(e,"string"==typeof n?n.length:0)),{value:n}),maxLength:(e,n)=>Object.assign(Object.assign({},t.Constraints.maximum(e,"string"==typeof n?n.length:0)),{value:n}),pattern:(e,t)=>{let n;return n="string"==typeof e?new RegExp(e):e,{valid:n.test(t),value:t}},required:(e,t)=>({valid:!e||null!=t&&""!==t,value:t}),enum:(e,t)=>({valid:e.indexOf(t)>-1,value:t})}}},t={};function n(r){var i=t[r];if(void 0!==i)return i.exports;var s=t[r]={exports:{}};return e[r].call(s.exports,s,s.exports,n),s.exports}n.d=(e,t)=>{for(var r in t)n.o(t,r)&&!n.o(e,r)&&Object.defineProperty(e,r,{enumerable:!0,get:t[r]})},n.o=(e,t)=>Object.prototype.hasOwnProperty.call(e,t),n.r=e=>{"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})};var r={};(()=>{"use strict";n.r(r),n.d(r,{Constants:()=>e,FormField:()=>s,Utils:()=>t,createFormInstance:()=>i.createFormInstance});const e={NS:"cmp",FORM_CONTAINER_INITIALISED:"AF_FormContainerInitialised"};class t{static readData(t,n){const r=t.dataset;let i={};const s=["is","hook"+(n.charAt(0).toUpperCase()+n.substring(1))];for(let t in r)if(Object.prototype.hasOwnProperty.call(r,t)){let n=r[t];0===t.indexOf(e.NS)&&(t=t.slice(e.NS.length),t=t.charAt(0).toLowerCase()+t.substring(1),-1===s.indexOf(t)&&(i[t]=n))}return i}static registerMutationObserver(e){let t=window.MutationObserver||window.WebKitMutationObserver||window.MozMutationObserver,n=document.querySelector("body");new t((function(t){t.forEach((function(t){var n=[].slice.call(t.addedNodes);n.length>0&&n.forEach((function(t){t.querySelectorAll&&[].slice.call(t.querySelectorAll(e.selectors.self)).forEach((function(t){let n=FormView.readData(t,e.IS).adaptiveformcontainerPath,r=window.af.formsRuntime.view.formContainer[n],i=new e({element:t});r.addField(i)}))}))}))})).observe(n,{subtree:!0,childList:!0,characterData:!0})}static getJson(e){return new Promise((t=>{let n=new XMLHttpRequest;n.open("GET",e,!0),n.responseType="json",n.onload=function(){let e=n.status;t(200===e?n.response:null)},n.send()}))}}var i=n(289);class s{constructor(e){this.formContainer=e.formContainer,this.parent=e.parent||e.formContainer,this.element=e.element,this.options=t.readData(this.element,this.getClass()),this.setId(this.element.id),this.bindEventListeners()}setId(e){this.id=e}getFormContainerPath(){return this.options.adaptiveformcontainerPath}getId(){return this.id}bindEventListeners(){this.element.addEventListener("change",(e=>{this._model.value=e.target.value}))}setModel(e){this._model=e}getModel(){return this._model}getClass(){throw new Error("Not Implemented")}getTagName(){throw new Error("Not Implemented")}setValue(e){throw new Error("Not implemented")}subscribe(){this._model.subscribe((e=>{let t=e.target.getState();t.valid?this.setValue(t.value):(alert(t.errorMessage),this.setValue(null))}))}}window.af={formsRuntime:{model:{form:{}},view:{formContainer:{},utils:{}},events:{}}}})(),window.FormView=r})();
```

Comment:
do we need to checkin this file ? I had the assumption it will be generated during build

Reply:
yes, no need to checking-in. I need to add it in gitignore

Reply:
I think we should delete this file and add in .gitignore